core/src/main/java/com/exadel/core/servlets/LikesServlet.java

```diff
@@ -0,0 +1,82 @@
+package com.exadel.core.servlets;
+
+import com.exadel.core.services.LikesService;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+import org.apache.sling.servlets.annotations.SlingServletName;
+import org.apache.sling.servlets.annotations.SlingServletPaths;
+import org.json.JSONObject;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import java.io.IOException;
+
+@Slf4j
+@Component(service = Servlet.class)
+@SlingServletPaths(value = "/json/likes")
+@SlingServletName(servletName = "LikesServlet")
+public class LikesServlet extends SlingAllMethodsServlet {
+
+    @Reference
+    private LikesService likesService;
+
+    /**
+     * Handle POST request, process likes/dislikes
+     * @param request SlingHttpServletRequest
+     * @param response SlingHttpServletResponse
+     * @throws ServletException throws servlet exception
+     * @throws IOException throws IO exception
+     */
+    @Override
+    protected void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response) throws ServletException, IOException {
+        try {
+            // Get request parameters
+            String product = request.getParameter("product");
+            boolean likeType = Boolean.parseBoolean(request.getParameter("like_type"));
+            String userIpAddress = getUserIp(request);
+
+            // Process like/dislike persistence
```

Comment:
It's better to move such logic into service

Reply:
Should I move getUserIp method into different service?